source/acquaman/IDEAS/IDEAS2DScanActionController.cpp

```diff
@@ -0,0 +1,199 @@
+#include "IDEAS2DScanActionController.h"
+
+#include "application/IDEAS/IDEAS.h"
+#include "actions3/AMActionSupport.h"
+#include "actions3/actions/AMWaitAction.h"
+#include "application/AMAppControllerSupport.h"
+#include "actions3/AMListAction3.h"
+#include "actions3/AMActionSupport.h"
+#include "dataman/AM2DScan.h"
+#include "beamline/IDEAS/IDEASBeamline.h"
+#include "analysis/AM2DNormalizationAB.h"
+#include "dataman/export/AMExporter2DAscii.h"
+#include "dataman/export/AMSMAKExporter.h"
+#include "dataman/export/AMExporterOptionSMAK.h"
+
+IDEAS2DScanActionController::IDEAS2DScanActionController(IDEAS2DScanConfiguration *configuration, QObject *parent)
+	: AMStepScanActionController(configuration, parent)
+{
+	configuration_ = configuration;
+
+	scan_ = new AM2DScan();
+	scan_->setScanConfiguration(configuration_);
+	scan_->setFileFormat("amCDFv1");
+	scan_->setIndexType("fileSystem");
+
+	// SMAK format requires a specific spectra file format.
+	AMExporterOptionSMAK *exportOptions = IDEAS::buildSMAKExporterOption("IDEAS2DDefault", true, true);
+	if(exportOptions->id() > 0)
+		AMAppControllerSupport::registerClass<IDEAS2DScanConfiguration, AMSMAKExporter, AMExporterOptionSMAK>(exportOptions->id());
+
+	QString scanName;
+
+	if(configuration_->userScanName() == ""){
+
+		scanName = configuration_->autoScanName();
+		scan_->setName(QString("%1").arg(scanName));
+	}
+
+	else{
+
+		scanName = configuration_->userScanName();
+		scan_->setName(QString("%1").arg(scanName));
+	}
+
+	int yPoints = configuration_->scanAxisAt(1)->numberOfPoints();
+
+	AMControlInfoList list;
+	list.append(AMBeamline::bl()->exposedControlByName("Sample Platform Vertical")->toInfo());
+	list.append(AMBeamline::bl()->exposedControlByName("Sample Platform Horizontal")->toInfo());
+	configuration_->setAxisControlInfos(list);
```

Comment:
How hard would it be to make these select-able, if I had two sample stages (or just have 2D scan controllers/configs/views)?   It just occurred to me that I might be able to mount the vacuum cross sample chamber (which has a vertical stage), on the old horizontal stage so that samples under vacuum can be 2D scanned with "tender" X-rays...